diff --git a/node_modules/@mathjax/src/cjs/output/svg/DefaultFont.js b/node_modules/@mathjax/src/cjs/output/svg/DefaultFont.js
index 153d11e..918635d 100644
--- a/node_modules/@mathjax/src/cjs/output/svg/DefaultFont.js
+++ b/node_modules/@mathjax/src/cjs/output/svg/DefaultFont.js
@@ -1,7 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.DefaultFont = exports.fontName = void 0;
-var default_js_1 = require("#default-font/svg/default.js");
+var default_js_1 = require("@mathjax/mathjax-newcm-font/js/svg/default.js");
 exports.fontName = default_js_1.Font.fontName;
 exports.DefaultFont = default_js_1.Font.DefaultFont;
 //# sourceMappingURL=DefaultFont.js.map
\ No newline at end of file
diff --git a/node_modules/@mathjax/src/cjs/util/context.js b/node_modules/@mathjax/src/cjs/util/context.js
index 26f1f25..cbbf417 100644
--- a/node_modules/@mathjax/src/cjs/util/context.js
+++ b/node_modules/@mathjax/src/cjs/util/context.js
@@ -36,7 +36,7 @@ exports.context = {
     os: (function () {
         var e_1, _a;
         if (exports.hasWindow && window.navigator) {
-            var app = window.navigator.appVersion;
+            var app = window.navigator.appVersion || '';
             var osNames = [
                 ['Win', 'Windows'],
                 ['Mac', 'MacOS'],
@@ -58,7 +58,7 @@ exports.context = {
                 }
                 finally { if (e_1) throw e_1.error; }
             }
-            if (window.navigator.userAgent.includes('Android')) {
+            if ((window.navigator.userAgent || '').includes('Android')) {
                 return 'Unix';
             }
         }
diff --git a/node_modules/@mathjax/src/mjs/output/svg/DefaultFont.js b/node_modules/@mathjax/src/mjs/output/svg/DefaultFont.js
index 7ef5885..7981509 100644
--- a/node_modules/@mathjax/src/mjs/output/svg/DefaultFont.js
+++ b/node_modules/@mathjax/src/mjs/output/svg/DefaultFont.js
@@ -1,4 +1,4 @@
-import { Font } from '#default-font/svg/default.js';
+import { Font } from '@mathjax/mathjax-newcm-font/js/svg/default.js';
 export const fontName = Font.fontName;
 export const DefaultFont = Font.DefaultFont;
 //# sourceMappingURL=DefaultFont.js.map
\ No newline at end of file
diff --git a/node_modules/@mathjax/src/mjs/util/context.js b/node_modules/@mathjax/src/mjs/util/context.js
index 969bdee..06f6f21 100644
--- a/node_modules/@mathjax/src/mjs/util/context.js
+++ b/node_modules/@mathjax/src/mjs/util/context.js
@@ -5,7 +5,7 @@ export const context = {
     document: hasWindow ? window.document : null,
     os: (() => {
         if (hasWindow && window.navigator) {
-            const app = window.navigator.appVersion;
+            const app = window.navigator.appVersion || '';
             const osNames = [
                 ['Win', 'Windows'],
                 ['Mac', 'MacOS'],
@@ -17,7 +17,7 @@ export const context = {
                     return os;
                 }
             }
-            if (window.navigator.userAgent.includes('Android')) {
+            if ((window.navigator.userAgent || '').includes('Android')) {
                 return 'Unix';
             }
         }
