diff --git a/node_modules/react-native-render-html/lib/commonjs/context/ListStyleSpecsProvider.js b/node_modules/react-native-render-html/lib/commonjs/context/ListStyleSpecsProvider.js
index ffd3e3c..4914232 100644
--- a/node_modules/react-native-render-html/lib/commonjs/context/ListStyleSpecsProvider.js
+++ b/node_modules/react-native-render-html/lib/commonjs/context/ListStyleSpecsProvider.js
@@ -6,7 +6,7 @@ Object.defineProperty(exports, "__esModule", {
 exports.useListStyleSpecs = useListStyleSpecs;
 exports.default = ListStyleSpecsProvider;
 
-var _ramda = require("ramda");
+var _mapObjIndexed = _interopRequireDefault(require("ramda/src/mapObjIndexed"));
 
 var _react = _interopRequireWildcard(require("react"));
 
@@ -56,7 +56,7 @@ function createSymbolicMarkerRenderer({
   };
 }
 
-const makeMarkerRenderers = (0, _ramda.mapObjIndexed)(value => {
+const makeMarkerRenderers = (0, _mapObjIndexed.default)(value => {
   if (value.type === 'unitary') {
     return { ...value,
       renderMarker: createSymbolicMarkerRenderer(value)
diff --git a/node_modules/react-native-render-html/src/context/ListStyleSpecsProvider.tsx b/node_modules/react-native-render-html/src/context/ListStyleSpecsProvider.tsx
index 822a4b7..17226fc 100644
--- a/node_modules/react-native-render-html/src/context/ListStyleSpecsProvider.tsx
+++ b/node_modules/react-native-render-html/src/context/ListStyleSpecsProvider.tsx
@@ -1,6 +1,6 @@
 /* eslint-disable react-native/no-inline-styles */
 import { MarkerBoxProps } from '@jsamr/react-native-li';
-import { mapObjIndexed } from 'ramda';
+import mapObjIndexed from 'ramda/src/mapObjIndexed';
 import React, {
   createContext,
   PropsWithChildren,
